rumqttc/src/lib.rs

```diff
@@ -247,6 +250,8 @@ pub struct MqttOptions {
     last_will: Option<LastWill>,
     /// Key type for TLS 
     key_type: Key,
+    /// Injected rustls ClientConfig for TLS, to allow more customisation.
+    tls_client_config: Option<Arc<ClientConfig>>,
```

Comment:
If the `ClientConfig` is shared (as per my comment above) then it can't safely be mutated, so the only safe way to do this is by replacing it with a new one.

I guess this wouldn't happen very often in practice though? What use case do you have in mind, if configuration is changed while the app is running and needs to be reloaded without restarting?

Reply:
I remember someone asking about using a different ca during connection failure. But they can just create a new `ClientConfig` I guess. So not a big deal